src/dbusutils.c

```diff
@@ -6,16 +6,134 @@
  **********************************************************************/
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
 #include "dbusutils.h"
 
 bool dbusutils_mainloop_running = false;
 
+void dbusutils_iter_append_string (DBusMessageIter *iter, int type, const char* string)
+{
+  char *tmp = strdup (string);
```

Comment:
Question: Worth Null checking string before strdup?

Reply:
Yeah cant hurt, Ill add that in